raianalysis/raianalysis/raianalyzer/raianalyzer.py

```diff
@@ -0,0 +1,49 @@
+# Copyright (c) Microsoft Corporation
+# Licensed under the MIT License.
+
+"""Defines the RAIAnalyzer class."""
+
+class RAIAnalyzer(object):
+    def __init__(self, model, train, test, target_column, transformation, task_type):
+        self.model = model
+        self.train = train
+        self.test = test
+        self.target_column = target_column
+        self.transformation = transformation
+        self.task_type = task_type
+
+    def add_fairness(self, sensitive_features):
+        pass
+
+    def add_explainer(self):
+        pass
+
+    def add_causal(self):
+        pass
+
+    def add_counterfactual(self,
+                           continuous_features=None,
+                           outcome_name=None,
+                           total_CFs=None, 
+                           desired_range=None,
+                           desired_class="opposite",
+                           permitted_range=None, 
+                           features_to_vary=None,
+                           method=None):
+        pass
+
+    def add_erroranalysis(self):
+        pass
+
+    def compute(self):
+        pass
+
+    def list(self):
```

Comment:
What exactly are we listing? The things pending to compute?

Reply:
great question, my question too!

Reply:
we discussed, it will be a dictionary of properties like on ExplanationClient list_model_explanations